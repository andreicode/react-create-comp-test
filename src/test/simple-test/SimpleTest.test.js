import React from 'react';
import { render, screen } from '@testing-library/react';
import SimpleTest from './SimpleTest';

describe('[COMPONENT] SimpleTest', () => {

  test('calling render with the same component on the same container does not remount', () => {
    const { getByTestId, rerender } = render(<SimpleTest number={1} />)
    expect(getByTestId('number-display').textContent).toBe('1')

    // re-render the same component with different props
    rerender(<SimpleTest number={2} />)
    expect(getByTestId('number-display').textContent).toBe('2')

    expect(getByTestId('instance-id').textContent).toBe('1')
  })

  it('renders correctly', () => {
    const { container } = render(<SimpleTest />);
    expect(container).toMatchSnapshot();
  });

  it('disabled button', () => {
    render(<SimpleTest />);
    expect(screen.getByTestId('disabledBtn').hasAttribute('disabled')).toBe(true)
  })

});
