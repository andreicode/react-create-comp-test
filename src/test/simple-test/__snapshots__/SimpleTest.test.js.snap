// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] SimpleTest renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <p>
      SimpleTest
    </p>
    Display number: 
    <span
      data-testid="number-display"
    />
    <br />
    Instance id: 
    <span
      data-testid="instance-id"
    >
      2
    </span>
    <br />
    <button
      data-testid="disabledBtn"
      disabled=""
    >
      Cannot click me
    </button>
    <br />
  </div>
</div>
`;
